gateway/reverse_proxy.go

```diff
@@ -679,7 +681,10 @@ func httpTransport(timeOut float64, rw http.ResponseWriter, req *http.Request, p
 		http2.ConfigureTransport(transport)
 	}
 
-	if p.TykAPISpec.Protocol == "h2c" {
+	p.logger.Debug("Out request url: ", outReq.URL.String())
+
+	if outReq.URL.Scheme == "h2c" {
```

Comment:
On transport create we just check if our target is h2c. 
As mentioned above it now works with load balancing too.